core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -217,6 +219,38 @@ final case class NonEmptyList[+A](head: A, tail: List[A]) {
     NonEmptyList((head, 0), bldr.result)
   }
 
+  /**
+    * Sorts this `NonEmptyList` according to an `Order` on transformed `B` from `A`
+    *
+    * {{{
+    * scala> import cats.data.NonEmptyList
+    * scala> import cats.instances.int._
+    * scala> val nel = NonEmptyList.of(('a', 4), ('z', 1), ('e', 22))
+    * scala> nel.sortBy(_._2)
+    * res0: cats.data.NonEmptyList[(Char, Int)] = NonEmptyList((z,1), (a,4), (e,22))
+    * }}}
+    */
+  def sortBy[B: Order](f: A => B): NonEmptyList[A] = {
+    toList.sortBy(f)(Order[B].toOrdering) match {
+      case x :: xs => NonEmptyList(x, xs)
+      case Nil     => sys.error("absurd")
+    }
+  }
+
+  /**
+    * Sorts this `NonEmptyList` according to an `Order`
+    *
+    * {{{
+    * scala> import cats.data.NonEmptyList
+    * scala> import cats.instances.int._
+    * scala> val nel = NonEmptyList.of(12, 4, 3, 9)
+    * scala> nel.sorted
+    * res0: cats.data.NonEmptyList[Int] = NonEmptyList(3, 4, 9, 12)
+    * }}}
+    */
+  def sorted[AA >: A](implicit AA: Order[AA]): NonEmptyList[AA] =
+    sortBy(a => a: AA)
```

Comment:
I'd rather not pay the cost to apply the identity function here. Can we instead directly implement?